src/types/markers.js

```diff
@@ -577,26 +580,37 @@ export type NativeAllocationPayload_Gecko = {|
   threadId?: number,
 |};
 
-export type IPCMarkerPayload = {|
+export type IPCMarkerPayload_Gecko = {|
   type: 'IPC',
   startTime: Milliseconds,
-  sendStartTime?: Milliseconds,
-  sendEndTime?: Milliseconds,
-  recvEndTime?: Milliseconds,
   endTime: Milliseconds,
   otherPid: number,
-  sendTid?: number,
-  recvTid?: number,
-  sendThreadName?: string,
-  recvThreadName?: string,
   messageType: string,
   messageSeqno: number,
   side: 'parent' | 'child',
   direction: 'sending' | 'receiving',
-  phase: 'endpoint' | 'transferBegin' | 'transferEnd',
+  // Phase is not present in older profiles (in this case the phase is "endpoint").
+  phase?: 'endpoint' | 'transferBegin' | 'transferEnd',
   sync: boolean,
 |};
 
+export type IPCMarkerPayload = {|
+  ...IPCMarkerPayload_Gecko,
+
+  // These fields are added in the deriving process from `IPCSharedData`, and
+  // correspond to data from all the markers associated with a particular IPC
+  // message.
+  startTime?: Milliseconds,
```

Comment:
Is `startTime` really optional? In the Gecko payload it's not marked as optional.

Reply:
Maybe I'm storing this data wrong, but after all the markers a single IPC message are correlated together, `startTime` is the timestamp of the first marker for that IPC message (i.e. the marker with direction = `sending` and phase = `endpoint`). Since that marker might not have been recorded, this time could be `undefined`.

However, in `deriveMarkersFromRawMarkerTable`, we set `start` and `dur` for all the markers (my understanding is that *these* fields are the ones that the timeline uses for rendering the markers). If we didn't record the first marker for our IPC message, `start` will be the timestamp for the *current* message and `dur` will be 0.

I think this logic is ok (I didn't see any issues in the UI at least), but maybe it'd be better to change things so we can't have undefined start/end times...